Content.Shared/Clothing/EntitySystems/StealthClothingSystem.cs

```diff
@@ -0,0 +1,136 @@
+using Content.Shared.Actions;
+using Content.Shared.Clothing.Components;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Stealth;
+using Content.Shared.Stealth.Components;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Clothing.EntitySystems;
+
+/// <summary>
+/// Handles the toggle action and disables stealth when clothing is unequipped.
+/// </summary>
+public sealed class StealthClothingSystem : EntitySystem
+{
+    [Dependency] private readonly SharedStealthSystem _stealth = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<StealthClothingComponent, GetItemActionsEvent>(OnGetItemActions);
+        SubscribeLocalEvent<StealthClothingComponent, ToggleStealthEvent>(OnToggleStealth);
+        SubscribeLocalEvent<StealthClothingComponent, AfterAutoHandleStateEvent>(OnHandleState);
+        SubscribeLocalEvent<StealthClothingComponent, GotUnequippedEvent>(OnUnequipped);
+    }
+
+    /// <summary>
+    /// Sets the clothing's stealth effect for the user.
+    /// </summary>
+    /// <returns>True if it was changed, false otherwise</returns>
+    public bool SetEnabled(EntityUid uid, EntityUid user, bool enabled, StealthClothingComponent? comp = null)
+    {
+        if (!Resolve(uid, ref comp) || comp.Enabled == enabled)
+            return false;
+
+        // prevent debug assert when ending round and its disabled
+        if (MetaData(user).EntityLifeStage >= EntityLifeStage.Terminating)
+            return false;
+
+        comp.Enabled = enabled;
+        Dirty(comp);
```

Comment:
```suggestion
        Dirty(uid, comp);
```

I am forbibben from adding more obsolete warnings.

Reply:
idk how that happen pro